### PR TITLE
Add `serde` and `Display` for `partiql-logical` and `serde` for `partiql-value`

### PR DESCRIPTION
**Pull request by am357**
*Description of changes:*

1. Adds the remaining `serde` features.
2. Adds an initial `Display` for `LogicalPlan`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-logical/src/lib.rs [hunk 1]
```diff
@@ -162,9 +163,27 @@ impl OpId {
...
+where
+    T: Default + Debug,
+{
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let flows = self.flows();
+        writeln!(f, "LogicalPlan")?;
+        writeln!(f, "---")?;
+        for (s, d, _w) in flows {
+            let src_node = self.operator(*s).unwrap();
+            let dst_node = self.operator(*d).unwrap();
```
Comment (alancai98) [inline partiql-logical/src/lib.rs:176]: nit: could provide error messages using `expect` rather than `unwrap`

File: partiql-value/src/lib.rs [hunk 2]
```diff
@@ -4,7 +4,7 @@ use std::cmp::Ordering;
 
 use std::borrow::Cow;
 use std::collections::HashSet;
-use std::fmt::{Debug, Formatter};
+use std::fmt::{Debug, Display, Formatter};
```
Comment (alancai98) [inline partiql-value/src/lib.rs:7]: `Display`'s an unused import

**Review by alancai98 — APPROVED** [review]
Looks good. Left a comment about an unused import and nit about `unwrap`

File: partiql-logical/src/lib.rs (hunk 1 above)
Comment (am357) [reply partiql-logical/src/lib.rs:176]: fixed.

**Pull request merged**